File: app/src/main/java/de/westnordost/streetcomplete/quests/crossing_type/AddCrossingTypeForm.java
package de.westnordost.streetcomplete.quests.crossing_type;

import de.westnordost.streetcomplete.R;
import de.westnordost.streetcomplete.quests.ImageListQuestAnswerFragment;
import de.westnordost.streetcomplete.view.Item;

public class AddCrossingTypeForm extends ImageListQuestAnswerFragment
{
    private final Item[] TYPES = new Item[] {
            new Item("traffic_signals", R.drawable.crossing_type_signals, R.string.quest_crossing_type_signals),
            new Item("uncontrolled", R.drawable.crossing_type_zebra, R.string.quest_crossing_type_uncontrolled),
            new Item("unmarked", R.drawable.crossing_type_unmarked, R.string.quest_crossing_type_unmarked)
    };

    @Override protected Item[] getItems() { return TYPES; }
    @Override protected int getItemsPerRow() { return 3; }
}

File: app/src/main/java/de/westnordost/streetcomplete/quests/crossing_type/AddCrossingTypeForm.kt
package de.westnordost.streetcomplete.quests.crossing_type

import de.westnordost.streetcomplete.quests.AImageListQuestForm

class AddCrossingTypeForm : AImageListQuestForm<CrossingType, CrossingType>() {

    override val items = CrossingType.entries.map { it.asItem() }
    override val itemsPerRow = 3

    override fun onClickOk(selectedItems: List<CrossingType>) {
        applyAnswer(selectedItems.single())
    }
}
